frontend/src/lib/hooks/useSearchPage.ts

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from 'react'
+import { useSearchParams } from 'react-router-dom'
+
+import logger from '../../utils/logger'
+import { loadData } from '../api'
+
+interface UseSearchPageOptions {
+  endpoint: string
+  pageTitle: string
+}
+
+interface UseSearchPageReturn<T> {
+  items: T[]
+  isLoaded: boolean
+  currentPage: number
+  totalPages: number
+  searchQuery: string
+  /* eslint-disable-next-line */
+  handleSearch: (query: string) => void
+  /* eslint-disable-next-line */
+  handlePageChange: (page: number) => void
+}
+
+export function useSearchPage<T>({
+  endpoint,
+  pageTitle,
+}: UseSearchPageOptions): UseSearchPageReturn<T> {
+  const [searchParams, setSearchParams] = useSearchParams()
+  const [items, setItems] = useState<T[]>([])
+  const [currentPage, setCurrentPage] = useState<number>(parseInt(searchParams.get('page') || '1'))
+  const [searchQuery, setSearchQuery] = useState<string>(searchParams.get('q') || '')
+  const [totalPages, setTotalPages] = useState<number>(0)
+  const [isLoaded, setIsLoaded] = useState<boolean>(false)
+
+  useEffect(() => {
+    const params = new URLSearchParams()
+    if (searchQuery) params.set('q', searchQuery)
+    if (currentPage > 1) params.set('page', currentPage.toString())
+    setSearchParams(params)
```

Comment:
Thanks for adding this! I didn't have a time to create an issue for that and now it's not needed.

Reply:
fixed the two warning